src/main/java/uk/gov/hmcts/ccd/domain/service/processor/SearchResultProcessor.java

```diff
@@ -0,0 +1,87 @@
+package uk.gov.hmcts.ccd.domain.service.processor;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.domain.model.definition.FieldType;
+import uk.gov.hmcts.ccd.domain.model.search.SearchResultView;
+import uk.gov.hmcts.ccd.domain.model.search.SearchResultViewColumn;
+import uk.gov.hmcts.ccd.domain.model.search.SearchResultViewItem;
+import uk.gov.hmcts.ccd.domain.types.CollectionValidator;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Component
+public class SearchResultProcessor {
```

Comment:
strictly speaking this is not generic. It's tied to date time processing. So better rename it to DateFormattingSearchResultProcessor?